src/main/java/com/digirati/taxonomy/manager/lookup/Lookup.java

```diff
@@ -0,0 +1,71 @@
+package com.digirati.taxonomy.manager.lookup;
+
+import com.digirati.taxonomy.manager.lookup.model.Match;
+
+import java.util.Collection;
+import java.util.stream.Collectors;
+
+/**
+ * Encapsulates all operations related to determining which concepts exist within a given piece of
+ * text.
+ */
+public class Lookup {
+
+    private AhoCorasickAutomaton ahoCorasickAutomaton;
+
+    private ConceptLookupTable conceptLookupTable;
+
+    /**
+     * Constructor creates all data structures necessary for extracting {@link
+     * com.digirati.taxonomy.manager.lookup.model.Concept}s from a given piece of text. At present
+     * this consists of an {@link AhoCorasickAutomaton} which is used to locate terms in text, and a
+     * {@link ConceptLookupTable} which is used to convert term matches into {@link
+     * com.digirati.taxonomy.manager.lookup.model.Concept} matches.
+     *
+     * @param terms all possible terms to look for
+     */
+    public Lookup(String... terms) {
+        ahoCorasickAutomaton = new AhoCorasickAutomaton(terms);
+        conceptLookupTable = new ConceptLookupTable();
+    }
+
+    public ConceptLookupTable getConceptLookupTable() {
+        return conceptLookupTable;
+    }
+
+    /**
+     * Uses the {@link AhoCorasickAutomaton} to locate any matching terms in the input text, and
+     * retrieves any {@link com.digirati.taxonomy.manager.lookup.model.Concept}s that these terms
+     * could represent.
+     *
+     * @param inputText the text from which to extract concepts
+     * @return a collection of {@link Match}es containing details of what terms were found, where
+     *     they were found, and what concepts these terms could be representing
+     */
+    public Collection<Match> search(String inputText) {
+        return ahoCorasickAutomaton.search(inputText).stream()
+                .map(emit -> new Match(emit, conceptLookupTable.get(emit.getKeyword())))
+                .collect(Collectors.toList());
+    }
+
+    /**
+     * Replaces the current {@link AhoCorasickAutomaton} with a new one. When changes are made to
+     * what terms should be listed in it, it isn't possible to update the existing automaton with
+     * those changes without a lot of very error-prone work - instead we simply rebuild it from
+     * scratch
+     *
+     * @param terms the terms to pass into the reconstructed automaton
+     */
+    public void rebuildAutomaton(String... terms) {
+        ahoCorasickAutomaton = new AhoCorasickAutomaton(terms);
```

Comment:
I don't think there will be, but I'm unsure of what form the data from which we build the lookup table will take - but it'll definitely need more than a collection of Strings. For now I figured I'd leave it as something for the calling code to do because if we were to add the call to clear the existing lookup table into this method we'd end up in the exact same state as merging these is trying to solve - half of the state would be correct and up to date and the other half wouldn't be. Once we've got an understanding of how that works, we should probably merge the two into one, but not until then.

Reply:
I think we can just remove these "rebuild" APIs and make those fields immutable.